boot-qemu.sh

```diff
@@ -177,24 +184,40 @@ function setup_qemu_args() {
     [[ -f ${KERNEL} ]] || die "${KERNEL} does not exist!"
 }
 
-
 # Invoke QEMU
 function invoke_qemu() {
     ${INTERACTIVE} || QEMU=( timeout "${TIMEOUT:=3m}" unbuffer "${QEMU[@]}" )
     if ${GDB:=false}; then
-        # Print message in bold green
-        printf '\033[01;32m'
-        echo
-        echo "Starting QEMU with GDB connection on port 1234..."
-        echo
-        echo "Use:"
-        echo
-        printf '\ttarget remote :1234\n'
-        echo
-        echo "to connect"
-        echo
-        printf '\033[0m'
-        QEMU=( "${QEMU[@]}" -s -S )
+        while true; do
+            if lsof -i:1234 &>/dev/null; then
+                red "Port :1234 already bound to. QEMU already running?"
+                exit 1
+            fi
+            green "Starting QEMU with GDB connection on port 1234..."
+            # Note: no -serial mon:stdio
+            "${QEMU[@]}" \
+              "${QEMU_ARCH_ARGS[@]}" \
+              -display none \
+              -initrd "${ROOTFS}" \
+              -kernel "${KERNEL}" \
+              -m "${QEMU_RAM:=512m}" \
+              -nodefaults \
+              -s -S &
```

Comment:
I couldn't figure a good way to conditionally add the `&`. Maybe there's a better way than this quick and dirty code duplication?

Reply:
This should be good enough. The only improvement I can think of is using a variable for common qemu parameters for debug/non-debug run.

Reply:
I do not think we can conditionally add the `&` but we could probably split the whole `${QEMU[@]} ...` invocation into another function like `run_qemu` or something like that then rename this function `do_testing` or something like that, since it is not longer JUST invoking QEMU.

No idea if that will work but might be worth testing.